mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/json/MimePart.java

```diff
@@ -120,7 +120,7 @@ public MimePart build() {
             Optional<ParsedContent> parsedContent = parseContent(textExtractor);
             return new MimePart(
                 headerCollectionBuilder.build(),
-                parsedContent.map(ParsedContent::getTextualContent),
+                parsedContent.map(ParsedContent::getTextualContent).map(Optional::get),
```

Comment:
IMO MimePart should accept an Optional

(Otherwize you throw everytime textual content is empty)